adapters/rubicon.go

```diff
@@ -266,9 +279,17 @@ func (a *RubiconAdapter) Call(ctx context.Context, req *pbs.PBSRequest, bidder *
 		if params.ZoneId == 0 {
 			return nil, errors.New("Missing zoneId param")
 		}
-		impExt := rubiconImpExt{RP: rubiconImpExtRP{ZoneID: params.ZoneId}}
+
+		impExt := rubiconImpExt{RP: rubiconImpExtRP{
+			ZoneID: params.ZoneId,
+			Target: params.Inventory,
+		}}
 		rubiReq.Imp[0].Ext, err = json.Marshal(&impExt)
 
+		// Amend the $.user object with $.user.ext.rp.target
+		userExt := rubiconUserExt{RP: rubiconUserExtRP{Target: params.Visitor}}
+		rubiReq.User.Ext, err = json.Marshal(&userExt)
```

Comment:
Since `rubiReq.User` is a pointer which is copied from `PBSRequest`, this makes a race condition if `req.App != nil`. See #140 for some similar fixes.

We're going to have to re-think this adapter-facing API so that these aren't so easy to introduce... but for now, copy the `User` object before mutating it.

Reply:
Great catch. Fixed!